### PR TITLE
399 Rename /programmes to /courses on api

### PR DESCRIPTION
**Pull request by bell-pa**
## Context

Trello: https://trello.com/c/DI1ijexV/399-rename-programmes-to-courses-on-api

## Changes in this PR

Update api.yml with new naming scheme following Trello card and this ER diagram: [courses-organisations-people-referrals.png](https://github.com/ministryofjustice/hmpps-accredited-programmes-api/blob/main/doc/architecture/diagrams/data-structures/courses-organisations-people-referrals.png)

### CODE REVIEW

**Review by yndajas — COMMENTED** [review]
Looks good, but should we also update the following?
- `doc/architecture/diagrams/data-structures/courses-organisations-people-referrals.wsd:17`
- `src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/controller/ProgrammesController.kt`
  - `ProgrammesController` -> `CoursesController`
  - `programmeType` -> `type`
  - `programmePrerequisites` -> `coursePrerequisites`
- `src/main/resources/static/api.yml`
  - `programmeType` -> `type`
  - `programmePrerequisites` -> `coursePrerequisites`

**Review by yndajas — COMMENTED** [review]
[duplicate comment]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/controller/ProgrammesController.kt [hunk 1]
```diff
@@ -2,47 +2,47 @@ package uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.controller
...
+        Course(
           id = UUID.randomUUID(),
           name = "Becoming new me +",
           programmeType = "TypeB",
           description = "Becoming new me + is for adult men and women with a medium or high risk of re offending...",
           programmePrerequisites = listOf(
-            ProgrammePrerequisite(key = "gender", value = "female"),
-            ProgrammePrerequisite(key = "risk score", value = "ORGS: 50+"),
-            ProgrammePrerequisite(key = "offence type", value = "some offence here"),
+            CoursePrerequisite(key = "gender", value = "female"),
```
Comment (Gweaton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/controller/ProgrammesController.kt:32]: Sorry, this wasn't in the ticket, but we've renamed `key` and `value` to `name` and `description` - is that something we could do here too?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/controller/ProgrammesController.kt [hunk 2]
```diff
@@ -2,47 +2,47 @@ package uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.controller
...
-import java.util.*
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.api.CoursesApiDelegate
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.api.model.Course
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.api.model.CoursePrerequisite
+import java.util.UUID
 
 @Service
-class ProgrammesController : ProgrammesApiDelegate {
-  override fun programmesGet(): ResponseEntity<List<Programme>> {
+class ProgrammesController : CoursesApiDelegate {
```
Comment (Gweaton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/controller/ProgrammesController.kt:11]: I think Ynda's mentioned this too, but let's rename this to `CoursesController`

**Review by Gweaton — COMMENTED** [review]
Thanks for this Peter, just a couple of small things if you wouldn't mind?

**Comment by bell-pa** [comment]
Some updates in response to comments from Ynda and George.

**Review by Gweaton — APPROVED** [review]
Thanks @bell-pa do you think you could squash these two commits together as they're essentially doing the same thing (renaming the courses endpoint) so feel like they should be part of the same history? We're very happy to (responsibly!) force push on these projects. Let me know if you'd like a hand with this.

**Pull request merged**